src/modules/launcher/PowerLauncher/MainWindow.xaml.cs

```diff
@@ -110,18 +110,21 @@ private void SuggestionsList_PreviewMouseLeftButtonUp(object sender, MouseButton
 
         private void ViewModel_PropertyChanged(object sender, PropertyChangedEventArgs e)
         {
-            if (Visibility == System.Windows.Visibility.Visible)
+            if(e.PropertyName == nameof(MainViewModel.MainWindowVisibility)) 
             {
-                // Not called on first launch
-                // Additionally called when deactivated by clicking on screen  
-                UpdatePosition();
-                BringProcessToForeground();
-
-                if (!_viewModel.LastQuerySelected)
+                if (Visibility == System.Windows.Visibility.Visible)

```

Comment:
Why wasn't it working earlier without the `e.PropertyName` check?

Reply:
Because `Visibility == System.Windows.Visibility.Visible` condition was always true when launcher was visible and the else block, which updates the query text was never called.